cpp/tests/lists/contains_tests.cpp

```diff
@@ -575,6 +575,146 @@ TEST_F(ContainsTest, VectorTypeRelatedExceptions)
   }
 }
 
+template <typename T>
+struct TypedContainsNaNsTest : public ContainsTest {
+};
+
+TYPED_TEST_CASE(TypedContainsNaNsTest, FloatingPointTypes);
+
+template <typename T>
+T get_nan(const char* nan_contents)
+{
+  return std::nan(nan_contents);
```

Comment:
`std::numeric_limits<T>::quiet_NaN()` is preferred.

Reply:
This was a conscious choice. `quiet_NaN()` is where I started, based on what I saw in other tests. But `quiet_NaN()` returns a single bit representation.
By constructing NaN values with different values, we test that they are treated as equivalent. We should not inadvertently compare the values bit-wise.

Reply:
@jrhemstad, I'm resolving this suggestion. I trust that the reasoning above is acceptable.

I'm still working on resolving the `rep()` suggestion. I'll update when I have that done.